packages/elements/src/tree/managers/tree-node.ts

```diff
@@ -145,40 +145,69 @@ export class TreeNode<T extends TreeDataItem = TreeDataItem> {
    * Return checked state of the item.
    * Apart from checked & unchecked state of `selected` accessor,
    * this method supports indeterminate state for items that some but not all of their children are selected.
-   * @returns item checked state: `1` for checked, `0` for unchecked and `-1` for indeterminate
+   * @returns item checked state: CHECKED (1), UNCHECKED (0), INDETERMINATE (-1)
    */
   public getCheckedState(): CheckedState {
     return this.manager.getItemCheckedState(this.item);
   }
 
+  /**
+   * Returns all ancestors of the item.
+   * @returns An array of ancestors as Tree Node
+   */
   public getAncestors(): TreeNode<T>[] {
     const ancestors = this.manager.getItemAncestors(this.item);
     return ancestors.map((ancestor) => this.manager.getTreeNode(ancestor) as TreeNode<T>);
   }
 
+  /**
+   * Returns the children of the item.
+   * @returns An array of children as Tree Node
+   */
   public getChildren(): TreeNode<T>[] {
     const children = this.manager.getItemChildren(this.item);
     return children.map((child) => this.manager.getTreeNode(child) as TreeNode<T>);
   }
 
-  public getDescendants(): TreeNode<T>[] {
-    const descendants = this.manager.getItemDescendants(this.item);
+  /**
+   * Returns all descendants of the item.
+   * @param depth Depth to retrieve
+   * @returns An array of descendants as Tree Node
+   */
+  public getDescendants(depth?: number): TreeNode<T>[] {
+    const descendants = this.manager.getItemDescendants(this.item, depth);
     return descendants.map((descendant) => this.manager.getTreeNode(descendant) as TreeNode<T>);
   }
 
+  /**
+   * Returns the parent of the item, if it has one.
+   * @returns Item parent as Tree Node or `null`
+   */
   public getParent(): TreeNode<T> | null {
     const parent = this.manager.getItemParent(this.item);
     return parent ? this.manager.getTreeNode(parent) : null;
   }
 
+  /**
+   * Returns whether the selected state of the item can be changed or not.
+   * @returns `True` if the item is not disabled or readonly
+   */
   public isSelectable(): boolean {
     return this.manager.isItemCheckable(this.item);
   }
 
+  /**
+   * Returns whether the item contains any children or not.
+   * @returns `True` if the item has children
+   */
   public isParent(): boolean {
     return this.manager.isItemParent(this.item);
   }
 
+  /**
+   * Returns whether the item has a parent or not.
+   * @returns `True` if the item has a parent
+   */
   public isChild(): boolean {
     return this.manager.isItemChild(this.item);
   }
```

Comment:
Let's update `rerender` JSDoc desciption to the same one with Tre Manager.

> Requests the item to be rerendered manually. Typically, this is not required. The render is triggered automatically when item's properties are updated.

Reply:
Updated.